src/core/border_router/routing_manager.cpp

```diff
@@ -1708,39 +1675,45 @@ void RoutingManager::RxRaTracker::HandleRouterTimer(void)
 
     for (Router &router : mRouters)
     {
-        if (!router.IsReachable())
+        if (!router.ShouldCheckReachability() || (router.mTimeout > now))
```

Comment:
Given that the implementation of `ShouldCheckReachability` is quite an arbitrary definition of when to check reachability, it is mentally hard to resolve what "Should _Not_ Check Reachability" actually means. You have to look up the method below in order to find out.

Reply:
Hmm, maybe move the `router.mTimeout > now` to `ShouldCheckReachability(now)` as well and make it a "one time" semantic wrapper for this specific case. Of course it cannot be reused for the case in line 1494 then.

Reply:
The code's logic in `HandleRouterTimer()` (`mRouterTimer` callback) can be interpreted as follows:

1. Iterate over all routers.
2. For each router:
   - If reachability shouldn't be checked OR the timeout hasn't passed yet, skip to the next router (`continue`).
   - Otherwise (timeout has passed), proceed with further processing for this router.